src/fshtypes/ValueSetFilter.ts

```diff
@@ -0,0 +1,53 @@
+import { FshCode } from './FshCode';
+import { FshEntity } from './FshEntity';
+
+export abstract class ValueSetFilter extends FshEntity {
+  public system: string;
+  public property: string;
+
+  withSystem(system: string) {
+    this.system = system;
+    return this;
+  }
+
+  withProperty(property: string) {
+    this.property = property;
+    return this;
+  }
+}
```

Comment:
Here too, staying closer to the FHIR spec might simplify things.  Why not:
```ts
class ValueSetFilter extends FshEntity {
  public property: string;
  public op: '=' | 'is-a' | 'descendent-of' | 'is-not-a' | 'regex' // | etc.
  public value: string;
}
```

Reply:
Then we don't need all those subclasses -- and when we process it, it's probably easier to setup a `switch` statement on `op` than having to check `instanceof` against each subclass.

Reply:
Also... as it is currently written, where would the `value` go?